beacon-chain/blockchain/forkchoice/service.go

```diff
@@ -41,18 +41,20 @@ type Store struct {
 	checkpointStateLock  sync.Mutex
 	seenAtts             map[[32]byte]bool
 	seenAttsLock         sync.Mutex
+	shouldArchive        bool
```

Comment:
Maybe just `archive`, the additional of `should` feels extra there

```suggestion
	archieve        bool
```

Reply:
+1 This could use a comment too. It's not clear why a fork choice module should know whether or not the node is running archival mode or not... Is this the best place for that logic?

Reply:
we want to save the attestations in the event we run an archiver service. Otherwise there isn't any point to saving attestations during sync. Since we save attestations in fork-choice, I need to know if the archiver flag is set

Reply:
It feels like a violation of responsibilities. I don't have a helpful suggestion so consider my feedback on this optional

Reply:
I understand your concerns, let me think of a better way to do this. I will see if I can maybe set the archiver flag as a global.